content/v2.0/query-data/guides/monitor-states.md

```diff
@@ -0,0 +1,192 @@
+  - **Unit:** the unit of time (`1s` (by default), `1m`, `1h`) used to increment the state duration.
+
+        ```js
+          |> stateDuration(fn: (r) =>
+           r._column_to_search == "value_to_search_for",
+          column: "state_duration",
+          unit: 1s)
+        ```
+
+2. Run `stateDuration()` to search each point for the specified value:
+
+    - For the first point that evaluates `true`, the state duration is set to `0`. For each consecutive point that evaluates `true`, the state duration increases by the time interval between each consecutive point (in specified units).
+    - If the state is `false`, the state duration is reset to `-1`.
+
+### Example query with stateDuration()
+
+<!-- once we have multiple example queries,link to multiple examples here with sub-descriptors -->
+
+The following query searches the `doors` bucket over the past 5 minutes to find how many seconds a door has been `closed`.
+
+```js
+  from(bucket: "doors")
+  |> range(start: -5m)
+  |> stateDuration(fn: (r) =>
+  r._value == "closed",
+   column: "door_closed",
+   unit: 1s)
+```
+
+In this example, `door_closed` is the **State duration** column. If you write data to the `doors` bucket every minute, the state duration increases by `60s` for each consecutive point where `_value` is `closed`. If `_value` is not `closed`, the state duration is reset to `0`.
+
+#### Query results
+
+Results for the example query above may look like this (for simplicity, we've omitted the measurement, tag, and field columns):
+
+```bash
+_time                   _value        door_closed
+2019-10-26T17:39:16Z    closed        0
+2019-10-26T17:40:16Z    closed        60
+2019-10-26T17:41:16Z    closed        120
+2019-10-26T17:42:16Z    open          -1
+2019-10-26T17:43:16Z    closed        0
+2019-10-26T17:44:27Z    closed        60
+```
+
+## Count the number of consecutive states
+
+1. Use the `stateCount()` function and include the following information:
+
+  - **Column to search:** any tag key, tag value, field key, field value, or measurement.
+  - **Value:** to search for in the specified column.
+  - **State count column:** a new column to store the state count─the number of consecutive records in which the specified value exists.
+
+        ```js
+        |> stateCount(fn: (r) => 
+            r._column_to_search == "value_to_search_for",
+            column: "state_count"`
+          )
+          ```
+
+2. Run `stateCount()` to search each point for the specified value:
+
+    - For the first point that evaluates `true`, the state count is set to `1`. For each consecutive point that evaluates `true`, the state count increases by 1.
+    - If the state is `false`, the state count is reset to `-1`.
+
+### Example query with stateCount()
+
+The following query searches the `doors` bucket over the past 5 minutes to find how many points have been counted where `_value` is `closed`.
```

Comment:
The following query searches the `doors` bucket over the past 5 minutes and calculates how many points have `closed` as thier `_value` .